app/src/main/java/io/kuenzler/whatsappwebtogo/BlobConverter.java

```diff
@@ -0,0 +1,144 @@
+import android.widget.Toast;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.content.FileProvider;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.text.DateFormat;
+import java.util.Date;
+
+public class BlobConverter {
+    private Context context;
+    public BlobConverter(Context context) {
+        this.context = context;
+    }
+
+
+    @JavascriptInterface
+    public void getBase64FromBlobData(String base64Data) throws IOException {
+        convertBase64StringToPdfAndStoreIt(base64Data);
+    }
+    public static String getBase64StringFromBlobUrl(String blobUrl) {
+        if(blobUrl.startsWith("blob")){
+            return "javascript: var xhr = new XMLHttpRequest();" +
+                    "xhr.open('GET', '"+ blobUrl +"', true);" +
+                    "xhr.responseType = 'blob';" +
+                    "xhr.onload = function(e) {" +
+                    "    if (this.status == 200) {" +
+                    "        var blobPdf = this.response;" +
+                    "        var reader = new FileReader();" +
+                    "        reader.readAsDataURL(blobPdf);" +
+                    "        reader.onloadend = function() {" +
+                    "            base64data = reader.result;" +
+                    "            Android.getBase64FromBlobData(base64data);" +
+                    "        }" +
+                    "    }" +
+                    "};" +
+                    "xhr.send();";
+
+        }
+        return "javascript: console.log('It is not a Blob URL');";
+    }
+    private void convertBase64StringToPdfAndStoreIt(String base64PDf) throws IOException {
+        final int notificationId = 1;
+        String currentDateTime = DateFormat.getDateTimeInstance().format(new Date());
+        String[] strings = base64PDf.split(",");
+        String extension;
+        switch (strings[0]) {//check image's extension
+            case "data:image/jpeg;base64":
+                extension = ".jpeg";
+                break;
+            case "data:image/png;base64":
+                extension = ".png";
+                break;
+            case "data:video/mp4;base64":
+                extension = ".mp4";
+                break;
+            case "data:video/3gp;base64":
+                extension = ".3gp";
+                break;
+            case "data:video/avi;base64":
+                extension = ".avi";
+                break;
+            case "data:image/gif;base64":
+                extension = ".gif";
+                break;
+            case "data:application/pdf;base64":
+                extension = ".pdf";
+                break;
+            default://should write cases for more images types
+                extension = "jpg";
+                break;
+        }
+
+        final File dwldsPath = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)+" "+currentDateTime+"downloaded" + extension);

```

Comment:
Why the space in the filename? shouldn't it be File.separator()?
Also I'd call the file "WAWTG_"+currentDateTime+extension